The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26343835834/job/77550402516) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/77550402516) [(plain)](https://github.com/rust-lang/rust/commit/d746383665088df35dab84a3530d30b83ea1cce3/checks/77550402516/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [codegen] tests/codegen-llvm/hint/cold_path.rs stdout ----
------rustc stdout------------------------------

------rustc stderr------------------------------
error: the feature named `sse2` is not valid for this target
##[error]  --> /checkout/tests/codegen-llvm/hint/cold_path.rs:54:18
   |
54 | #[target_feature(enable = "sse2")]
   |                  ^^^^^^^^^^^^^^^ `sse2` is not valid for this target
   |
   = help: valid names are: `lse2`, `sme2`, `sve2`, `lse`, and `sha2` and 89 more

error: aborting due to 1 previous error


------------------------------------------

error: compilation failed!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/codegen-llvm/hint/cold_path.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "-O" "-Cdebug-assertions=no" "-Zcodegen-source-order" "--emit" "llvm-ir" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/hint/cold_path/cold_path.ll" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Copt-level=3"
stdout: none
--- stderr -------------------------------
error: the feature named `sse2` is not valid for this target
##[error]  --> /checkout/tests/codegen-llvm/hint/cold_path.rs:54:18
   |
54 | #[target_feature(enable = "sse2")]
   |                  ^^^^^^^^^^^^^^^ `sse2` is not valid for this target
   |
   = help: valid names are: `lse2`, `sme2`, `sve2`, `lse`, and `sha2` and 89 more

error: aborting due to 1 previous error
------------------------------------------

---- [codegen] tests/codegen-llvm/hint/cold_path.rs stdout end ----
```

</details>

--- a/library/core/src/hint.rs
+++ b/library/core/src/hint.rs
@@ -778,6 +778,9 @@ pub const fn unlikely(b: bool) -> bool {
 #[stable(feature = "cold_path", since = "1.95.0")]
 #[rustc_const_stable(feature = "cold_path", since = "1.95.0")]
 #[inline(always)]
+// Even if for some reason the cold_path intrinsic is not visible to codegen, the coldness will
+// ensure that branches this is in are still known to be cold.
+#[cold]
 pub const fn cold_path() {
     crate::intrinsics::cold_path()
 }

--- a/tests/codegen-llvm/hint/cold_path-target_feature.rs
+++ b/tests/codegen-llvm/hint/cold_path-target_feature.rs
@@ -1,0 +1,56 @@
+//@ compile-flags: -Copt-level=3
+//@ only-x86_64
+#![crate_type = "lib"]
+
+// This test checks that hint::cold_path still works in #[target_feature] functions.
+
+use std::hint::cold_path;
+
+#[inline(never)]
+#[no_mangle]
+pub fn path_a() {
+    println!("path a");
+}
+
+#[inline(never)]
+#[no_mangle]
+pub fn path_b() {
+    println!("path b");
+}
+
+#[no_mangle]
+pub fn test1(x: bool) {
+    if x {
+        path_a();
+    } else {
+        cold_path();
+        path_b();
+    }
+
+    // CHECK-LABEL: @test1(
+    // CHECK: br i1 %x, label %bb1, label %bb2, !prof ![[NUM:[0-9]+]]
+    // CHECK: bb2:
+    // CHECK: path_b
+    // CHECK: bb1:
+    // CHECK: path_a
+}
+
+#[no_mangle]
+#[target_feature(enable = "sse2")]
+pub fn with_target_feature(x: bool) {
+    if x {
+        path_a();
+    } else {
+        cold_path();
+        path_b();
+    }
+
+    // CHECK-LABEL: @with_target_feature(
+    // CHECK: br i1 %x, label %bb1, label %bb2, !prof ![[NUM]]
+    // CHECK: bb2:
+    // CHECK: path_b
+    // CHECK: bb1:
+    // CHECK: path_a
+}
+
+// CHECK: ![[NUM]] = !{!"branch_weights", {{(!"expected", )?}}i32 2000, i32 1}

